### PR TITLE
build: Parse version information in msvc-autogen.py

### LINKED ISSUE #23073: Auto-generate MSVC version information
Right now, bumping the version involves updating the version in two places:

- `configure.ac`
- `build_msvc/bitcoin_config.h` (in multiple places—see for example commit f277b1782c5aae0fab83e3d678189c150ae0f263)

I think it should be doable to parse the information from `configure.ac` in `msvc-autogen.py`, so that this is no longer something error-prone that has to be done manually.

If you do this, also update [the release process](https://github.com/bitcoin/bitcoin/blob/master/doc/release-process.md).

#### Useful skills:

Python, MSVC

#### Want to work on this issue?

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by CallMeMisterOwl**
Added function to msvc-autogen.py that automatically parses changes from 
configure.ac into bitcoin_config.h.
The function supports following values:
- CLIENT_VERSION_BUILD
- CLIENT_VERSION_IS_RELEASE
- CLIENT_VERSION_MAJOR
- CLIENT_VERSION_MINOR
- COPYRIGHT_YEAR
- PACKAGE_STRING
- PACKAGE_VERSION

### CODE REVIEW

**Comment by CallMeMisterOwl** [comment]
I tried to avoid using additional libraries e.g. parsers, I hope that's fine. 
Also if you're not satisfied with the code (style, lengths etc), please let me know.

**Comment by laanwj** [comment]
Thanks for having a go at #23073.

However, I don't see how this works. You're adding a function but not actually calling it?

File: build_msvc/msvc-autogen.py [hunk 1]
```diff
@@ -57,6 +58,38 @@ def set_common_properties(toolset):
     with open(os.path.join(SOURCE_DIR, '../build_msvc/common.init.vcxproj'), 'w', encoding='utf-8',newline='\n') as wfile:
         wfile.write(s)
 
+def parse_config_into_btc_conifg():
```
Comment (laanwj) [inline build_msvc/msvc-autogen.py:61]: typo `parse_config_into_btc_config`

**Comment by CallMeMisterOwl** [comment]
> However, I don't see how this works. You're adding a function but not actually calling it?
Of course your right, maybe I should focus before pushing random stuff. Sorry about that...

**Pull request closed without merging**